src/Plugin/EntityReferenceSelection/OgUserSelection.php

```diff
@@ -0,0 +1,193 @@
+    $this->membershipManager = $membership_manager;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public static function create(ContainerInterface $container, array $configuration, $plugin_id, $plugin_definition) {
+    return new static(
+      $configuration,
+      $plugin_id,
+      $plugin_definition,
+      $container->get('entity.manager'),
+      $container->get('module_handler'),
+      $container->get('current_user'),
+      $container->get('database'),
+      $container->get('og.membership_manager')
+    );
+  }
+
+  /**
+   * Get the selection handler of the field.
+   *
+   * @return Drupal\Core\Entity\Plugin\EntityReferenceSelection\DefaultSelection
+   *   Returns the selection handler.
+   */
+  public function getSelectionHandler() {
+    $options = [
+      'target_type' => 'user',
+    ];
+    return \Drupal::service('plugin.manager.entity_reference_selection')->getInstance($options);
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  protected function buildEntityQuery($match = NULL, $match_operator = 'CONTAINS') {
+    $query = parent::buildEntityQuery($match, $match_operator);
+
+    // Anon can't be a group member.
+    $query->condition('uid', 0, '<>');
+
+    // The user entity doesn't have a label column.
+    if (isset($match)) {
+      $query->condition('name', $match, $match_operator);
+    }
+
+    // Adding the permission check is sadly insufficient for users: core
+    // requires us to also know about the concept of 'blocked' and 'active'.
+    if (!$this->currentUser->hasPermission('administer users')) {
+      $query->condition('status', 1);
+    }
+
+    return $query;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function entityQueryAlter(SelectInterface $query) {
+    // Exclude users who are already in the current group.
+    // This has to be done on the SQL query rather than the entity query,
+    // because a reverse relationship to the OG membership entity is needed.
+
+    // @todo implement an easier, more consistent way to get the group type. At
+    // the moment, this works either for checkboxes or OG Autocomplete widget
+    // types on entities that have a getGroup() method. It also does not work
+    // properly every time; for example during validation.
+    $group = NULL;
+    if (isset($this->configuration['entity'])) {
+      $entity = $this->configuration['entity'];
+      $group = is_callable([$entity, 'getGroup']) ? $entity->getGroup() : NULL;
+    }
+
+    if (isset($this->configuration['handler_settings']['group'])) {
+      $group = $this->configuration['handler_settings']['group'];
+    }
+
+    if ($group === NULL) {
+      return $query;
+    }
+
+    // Left join to the OG membership base table.
+    $query->leftJoin('og_membership', 'ogm', "base_table.uid = ogm.uid");
+
+    // Exclude any users who are in the current group.
+    $query->condition('ogm.entity_type', $group->getEntityTypeId(), '!=');
+    $query->condition('ogm.entity_id', $group->id(), '!=');
```

Comment:
This results in the user autocomplete return no results. Will provide a patch shortly

Reply:
https://github.com/jludwig/og/pull/2

Reply:
Indeed. When commenting these lines it still triggers a UniqueOgMembershipConstraint when saving the form.

Reply:
Fixed in https://github.com/jludwig/og/pull/2/files